eis_toolkit/vector_processing/idw_interpolation.py

```diff
@@ -34,24 +35,37 @@ def _idw_interpolation(
     y = np.linspace(grid_y_min, grid_y_max, raster_height)
     y = y[::-1].reshape(-1, 1)
 
-    interpolated_values = _idw_core(points[:, 0], points[:, 1], values, x, y, power)
+    interpolated_values = _idw_core(points[:, 0], points[:, 1], values, x, y, power, search_radius)
     interpolated_values = interpolated_values.reshape(raster_height, raster_width)
 
     return interpolated_values
 
 
 #  Distance calculations
-def _idw_core(x, y, z, xi, yi: np.ndarray, power: Number) -> np.ndarray:
+def _idw_core(x, y, z, xi, yi: np.ndarray, power: Number, search_radius: Optional[Number]) -> np.ndarray:
     over = np.zeros((len(yi), len(xi)))
     under = np.zeros((len(yi), len(xi)))
     for n in range(len(x)):
         dist = np.hypot(xi - x[n], yi - y[n])
-        # Add a small epsilon to avoid division by zero
-        dist = np.where(dist == 0, 1e-12, dist)
-        dist = dist**power
 
-        over += z[n] / dist
-        under += 1.0 / dist
+        # Exclude points outside search radius
+        if search_radius is not None:
+            mask = dist <= search_radius
+            if not np.any(mask):
+                continue
+
+            # Add a small epsilon to avoid division by zero
+            dist = np.where(dist[mask] == 0, 1e-12, dist[mask]) ** power
+
+            over[mask] += z[n] / dist
+            under[mask] += 1.0 / dist
+
+        else:
+            # Add a small epsilon to avoid division by zero
+            dist = np.where(dist == 0, 1e-12, dist) ** power
+
+            over += z[n] / dist
+            under += 1.0 / dist
 
     interpolated_values = over / under
```

Comment:
Even though this seems to produce correct results, I can throw error sometimes. This should fix it
```suggestion
    interpolated_values = np.divide(over, under, out=np.full_like(over, np.nan), where=under != 0)
```